kiwi-desc-saltboot/saltboot/suse-SLES12/root/saltboot.sh

```diff
@@ -167,17 +158,36 @@ if [ -z "$CUR_MASTER" -o "salt" == "$CUR_MASTER" ] ; then
     if [ -z "$MASTER" ] ; then
         MASTER=`dig $DIG_OPTIONS -t CNAME salt.$DOMAIN | sed -e 's|;;.*||' -e 's|\.$||' `
     fi
-    # if it fails, do nothing and let it fall back to 'salt'
-    if [ -n "$MASTER" ] ; then
-        echo "master: $MASTER" > $INITRD_SALT_ETC/minion.d/master.conf
-    fi
-    # else
-    # ... if it fails, do nothing and let it fall back to 'salt'
 fi
 
 Echo "Using Salt master: ${MASTER:-$CUR_MASTER}"
 echo "Using Salt master: ${MASTER:-$CUR_MASTER}" > /progress
 
+if ! grep -q "^master: ${MASTER:-$CUR_MASTER}$" $INITRD_SALT_ETC/minion.d/susemanager.conf ; then
```

Comment:
The same as in dracut-saltboot

```suggestion
if ! grep -q "^master: ${MASTER:-$CUR_MASTER}$" $INITRD_SALT_ETC/minion.d/susemanager.conf 2>/dev/null; then
```